lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -232,6 +234,71 @@ def test_course_about_in_cart(self):
         self.assertEqual(response.status_code, 200)
         self.assertIn(in_cart_span, response.content)
 
+    def prepare_course_about_with_ecommerce(self, min_price=0):
+        """ Helper function to prepare an commerce configuration, course and request for tests. """
+        # Create an ecommerce configuration
+        checkout_page = '/test_basket/'
+        sku = 'TEST123'
+        CommerceConfiguration.objects.create(
+            checkout_on_ecommerce_service=True,
+            single_course_checkout_page=checkout_page
+        )
+        course = CourseFactory.create(org="new", number="unenrolled", display_name="course")
+        CourseModeFactory(mode_slug=CourseMode.PROFESSIONAL, course_id=course.id, sku=sku, min_price=min_price)
+        request = self.request_factory.get(reverse('about_course', args=[course.id.to_deprecated_string()]))
+        return checkout_page, sku, course, request
+
+    def test_ecommerce_checkout_with_anonymous(self):
+        checkout_page, sku, course, request = self.prepare_course_about_with_ecommerce()
+        request.user = AnonymousUser()
+        mako_middleware_process_request(request)
+        ecommerce_enrollment_link = (
```

Comment:
I ended up splitting the tests into two with DDT, because two of them test when the shopping cart setting is enabled: 559b964d9669b43b9ed4905b2e3ebf4cc77882d1